src/halmos/utils.py

```diff
@@ -1,9 +1,21 @@
 # SPDX-License-Identifier: AGPL-3.0
 
+import re
+
 from typing import Dict, Tuple
 
 from z3 import *
 
+def hexify(x):
+    if isinstance(x, str):
+        return re.sub(r'\b(\d+)\b', lambda match: hex(int(match.group(1))), x)
+    elif isinstance(x, int):
+        return hex(x)
+    elif is_bv_value(x):
+        return hex(x.as_long())
```

Comment:
I think it would be great to preserve the bitsize of the value in the hex output

```
# not this:
hex(BitVecVal(0x42, 160).as_long())
'0x42'

# this:
'0x' +  BitVecVal(0x42, 160).as_long().to_bytes(20, 'big').hex()
'0x0000000000000000000000000000000000000042'
```

Reply:
Using bv_value_to_bytes from the other PR, I would suggest the following:

```py
def hexify(x):
    if isinstance(x, str):
        return re.sub(r'\b(\d+)\b', lambda match: hex(int(match.group(1))), x)
    elif isinstance(x, int):
        return hex(x)
    elif isinstance(x, bytes):
        return '0x' + x.hex()
    elif is_bv_value(x):
        return hexify(bv_value_to_bytes(x))
    else:
        return hexify(str(x))
```

Reply:
good point! will do!